ansible/roles-infra/infra-ec2-template-generate/tasks/validate_instances_amis.yaml

```diff
@@ -0,0 +1,116 @@
+  - name: test 1
+    debug: msg="{{ instances }}"
+
+  - name: prep facts
+    set_fact:
+      instances_DoD: "{{ [] }}"  # DoD
+  
+  - name: fail if I can't map an image to aws_ami_region_mapping
+    loop: "{{ instances }}"
+    when:
+    - item.image is defined
+    - aws_ami_region_mapping[aws_region][item.image] is not defined
+    fail:
+      msg: >- 
+        instance.image "{{ item.image }}" is not defined in aws_ami_region_mapping.
+
+  - name: create instances_DoD from instances LoD
+    loop: "{{ instances }}"
+    loop_control:
+      extended: yes
+    set_fact: 
+      instances_DoD: >-
+        {{ 
+          instances_DoD| combine(
+            { ansible_loop.index0: instances[ansible_loop.index0]  }
+          )
+        }}
+
+  - name: test if bump RHEL<VERSION> images to GOLD, if available in aws_ami_region_mapping
+    loop: "{{ instances_DoD|dict2items }}"
```

Comment:
And then you wouln'dt need `dict2items`

Reply:
I turn the index of the list into a keybut I want to keep order.  Otherwise I can't find the instances in the proper order.  If order even matters.